ow_power_system/src/PowerSystemPack.cpp

```diff
@@ -0,0 +1,784 @@
+    for (int i = 0; i < NUM_NODES; i++)
+    {
+      double excessiveDraw = m_nodes[i].node.RunOnce();
+      // Add to the warning message if the draw exceeded set limits (which returns
+      // a value greater than 0).
+      if (excessiveDraw > 0)
+      {
+        // Formatting for the warning message, to display up to 4 nodes
+        // per line for conciseness.
+        std::string header = "N";
+        if (i < 10)
+        {
+          header += "0";
+        }
+        header += std::to_string(i) + ": " + std::to_string(excessiveDraw) + "W";
+        switch(msgIndex)
+        {
+          case 0:
+            if (firstLine)
+            {
+              firstLine = false;
+            }
+            else
+            {
+              highDrawMsg += ",\n";
+            }
+            highDrawMsg += header;
+            msgIndex++;
+            break;
+          case 1:
+            highDrawMsg += ", " + header;
+            msgIndex++;
+            break;
+          case 2:
+            highDrawMsg += ", " + header;
+            msgIndex++;
+            break;
+          case 3:
+            highDrawMsg += ", " + header;
+            msgIndex = 0;
+            break;
+          default:
+            break;
+        }
+      }
+
+      m_nodes[i].node.GetPowerStats(m_nodes[i].model.timestamp, m_nodes[i].model.wattage,
+                                    m_nodes[i].model.voltage, m_nodes[i].model.temperature);
+      
+      // If the timestamp is the same as the previous one (happens during startup),
+      // do not publish the data to prevent crashes.
+      if (m_nodes[i].previous_time != m_nodes[i].model.timestamp)
+      {
+        auto timestamp = START_TIME + std::chrono::milliseconds(
+          static_cast<unsigned>(m_nodes[i].model.timestamp * 1000)
+        );
+        
+        // Compile a vector<shared_ptr<DoubleMessage>> and then individually
+        // publish each component.
+        std::vector<std::shared_ptr<DoubleMessage>> data_to_publish;
+        double input_power, input_tmp, input_voltage;
+
+        // HACK ALERT (May 2023):
+        // The first batch of data (sent in during startup) seems to be lost to 
+        // the void. This is a bandaid fix to send the initial data in twice
+        // for each node on the next loop.
+        if (startLoops > 0)
+        {
+          // The very first values sent in should be the init values.
+          input_power = m_initial_power;
+          input_voltage = m_initial_voltage;
+          input_tmp = m_initial_temperature;
+          startLoops--;
+        }
+        else
+        {
+          input_power = m_nodes[i].model.wattage;
+          input_voltage = m_nodes[i].model.voltage;
+          input_tmp = m_nodes[i].model.temperature;
+        }
+
+        // DEBUG PRINT
+        if (m_inputs_print_debug && (m_nodes[i].model.timestamp > 0))
+        {
+          std::string header = "N";
+          if (i < 10)
+          {
+            header += "0";
+          }
```

Comment:
I simplified both cases to a single 'if' statement, and I'm not sure if that is worth making a separate function for it. However, it might be worth making standalone functions for each of the debug printouts rather than having them defined within the loop (since it's already pretty large). I'll go ahead and implement that.